### PR TITLE
goread: init at 1.6.5

### LINKED ISSUE #326820: Package request: goread
**Project description**

goread is an RSS/Atom feed reader for your terminal. It allows you to categorize and follow feeds and read articles right in the terminal.

**Metadata**

* homepage URL: https://github.com/TypicalAM/goread
* source URL: https://github.com/TypicalAM/goread
* license: gpl3
* platforms: unix, linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by meir**
## Description of changes

https://github.com/TypicalAM/goread

Terminal RSS/Atom feed reader made using golang and [bubbletea](https://github.com/charmbracelet/bubbletea).

Closes https://github.com/NixOS/nixpkgs/issues/326820

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/go/goread/package.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  meta = with lib; {
+    description =
+      "Beautiful program to read your RSS/Atom feeds right in the terminal!";
+    homepage = "https://github.com/TypicalAM/goread";
+    changelog = "https://github.com/TypicalAM/goread/releases/tag/v${version}";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ meir ];
+    mainProgram = "goread";
+  };
```
Comment (AndersonTorres) [inline pkgs/by-name/go/goread/package.nix:24]:
```suggestion
  meta = {
    description = "Beautiful program to read your RSS/Atom feeds right in the terminal";
    homepage = "https://github.com/TypicalAM/goread";
    changelog = "https://github.com/TypicalAM/goread/releases/tag/v${version}";
    license = lib.licenses.gpl3Plus;
    mainProgram = "goread";
    maintainers = with lib.maintainers; [ meir ];
  };
```

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -13164,6 +13164,11 @@
     githubId = 1788114;
     name = "Meghea Iulian";
   };
+  meir = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:13167]: 1. Add a reachable e-mail;
2. Split this into a separate commit (no need for a new PR).

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
P.S.: add a testVersion.

**Comment by meir** [comment]
> P.S.: add a testVersion.

im not familiar on how to add a testVersion, are there any official docs on how to properly implement this?

File: pkgs/by-name/go/goread/package.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    hash = "sha256-SRVXcdgtRpWqvO28CnUcx40nFJnG+Hd94Ezgaj5xK6A=";
+  };
+
+  vendorHash = "sha256-/kxEnw8l9S7WNMcPh1x7xqiQ3L61DSn6DCIvJlyrip0=";
+
+  meta = with lib; {
+    description = "Beautiful program to read your RSS/Atom feeds right in the terminal!";
+    homepage = "https://github.com/TypicalAM/goread";
+    changelog = "https://github.com/TypicalAM/goread/releases/tag/v${version}";
+    license = licenses.gpl3Plus;
```
Comment (AndersonTorres) [inline pkgs/by-name/go/goread/package.nix:20]:
```suggestion
    license = lib.licenses.gpl3Plus;
```

File: pkgs/by-name/go/goread/package.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    mainProgram = "goread";
+    maintainers = with maintainers; [ meir ];
```
Comment (AndersonTorres) [inline pkgs/by-name/go/goread/package.nix:22]:
```suggestion
    maintainers = with lib.maintainers; [ meir ];
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/go/goread/package.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchFromGitHub {
+    owner = "TypicalAM";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-SRVXcdgtRpWqvO28CnUcx40nFJnG+Hd94Ezgaj5xK6A=";
+  };
+
+  vendorHash = "sha256-/kxEnw8l9S7WNMcPh1x7xqiQ3L61DSn6DCIvJlyrip0=";
+
+  meta = with lib; {
```
[the remark this answers is not in the document]
Comment (meir) [reply pkgs/by-name/go/goread/package.nix:16]: is there a specific reason why you'd rather have it without this? 
looking through other packages and merge requests i don't see any specific standard for this used.

[the remark this answers is not in the document]
Comment (Aleksanaa) [reply pkgs/by-name/go/goread/package.nix:16]: Because https://github.com/NixOS/nixpkgs/issues/208242. Nixpkgs is not always a good standard, that's why we have to apply treewide changes from time to time.
